python/fusion_engine_client/parsers/mixed_log_reader.py

```diff
@@ -107,6 +116,14 @@ def __init__(self, input_file, warn_on_gaps: bool = False, show_progress: bool =
         self._original_index = fast_indexer.fast_generate_index(input_path, force_reindex=ignore_index,
                                                                 save_index=save_index, max_bytes=max_bytes)
         self.next_index_elem = 0
+        self.index = self._original_index
+        self.filtered_message_types = False
+
+        self.get_available_source_ids()
+        self.filter_in_place(None, source_ids=self.requested_source_ids)
+
```

Comment:
I suggest this blank and the line above and grouping all of the filter in place calls together